knative-operator/pkg/common/service_monitor.go

```diff
@@ -54,23 +78,171 @@ func SetupServiceMonitor(ctx context.Context, cfg *rest.Config, metricsPort int3
 // serveCRMetrics gets the Operator/CustomResource GVKs and generates metrics based on those types.
 // It serves those metrics on "http://metricsHost:operatorMetricsPort".
 func serveCRMetrics(cfg *rest.Config, metricsHost string, operatorMetricsPort int32) error {
-	// Below function returns filtered operator/CustomResource specific GVKs.
-	// For more control override the below GVK list with your own custom logic.
-	filteredGVK, err := k8sutil.GetGVKsFromAddToScheme(apis.AddToScheme)
+
+	// If we dont use a custom list here, the typical call to get a filtered list of gvks using k8sutil.GetGVKsFromAddToScheme(apis.AddToScheme)
+	// will end up with resources the Operator does not have access to eg. `Kind=LimitRange`
+	// The list of resources returned is unrelated to our purposes here, thus the customization.
+	gvkFilterList := []schema.GroupVersionKind{
+		schema.GroupVersionKind{
+			Group:   "operator.knative.dev",
+			Version: "v1alpha1",
+			Kind:    "KnativeServing",
+		},
+		schema.GroupVersionKind{
+			Group:   "operator.knative.dev",
+			Version: "v1alpha1",
+			Kind:    "KnativeEventing",
+		},
```

Comment:
Just wondering... do we have a JIRA to add (for 1.11+) some `KnativeKafka` ? 
/cc @aliok

Reply:
@matzew would be great as I checked the labels for the kafka source too. Hope this is generic enough to detect future sources .